src/plugins/plugin.colors.ts

```diff
@@ -0,0 +1,105 @@
+import type {Chart, ChartDataset} from '../types';
+
+export interface ColorsPluginOptions {
+  enabled?: boolean;
+}
+
+type DatasetColorizer = (dataset: ChartDataset, i: number) => void;
+
+interface ColorsDescriptor {
+  backgroundColor?: unknown;
+  borderColor?: unknown;
+}
+
+const BORDER_COLORS = [
```

Comment:
Could make sense to have an option to allow the user to use own colors scheme?

Reply:
Yes, but I think this will be requested in the future because the users prefer to have something out-of-the-box instead of having a dependency with other plugins. And having already done the 80% of the work, I think it will happen. Anyway, just my 2 cents.

Reply:
@stockiNail Adding options for the background alpha and a user-defined color scheme make a lot of sense! However, I don't have an intention to "circumvent" the color-related options that Chart.js have or provide a way to "bypass" them. 

Here's my reasoning: if anyone wants to have a custom palette, why wouldn't them just set `borderColor`/`backgroundColor` on their datasets, as suggested in the docs? I think that if this plugin tries to "compete" with the existing color-related options, it will definitely fail. So, I don't want event try that 😄